### PR TITLE
Add Jil JSON test

### PR DESCRIPTION
**Pull request by NickCraver**
This moves the JSON.Net test to /json/newtonsoft (to match the other multi-implementation test patterns) and adds /json/jil as another JSON serialization test.

### CODE REVIEW

File: src/Benchmarks/project.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "Microsoft.AspNet.Server.WebListener": "1.0.0-*",
     "Microsoft.AspNet.StaticFiles": "1.0.0-*",
     "Microsoft.Extensions.WebEncoders": "1.0.0-*",
-    "Newtonsoft.Json": "7.0.1"
+    "Newtonsoft.Json": "7.0.1",
+    "Jil": "2.13.0-*"
```
Comment (davidfowl) [inline src/Benchmarks/project.json]: Can we use a specific version here?

File: src/Benchmarks/JsonJilMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        public Task Invoke(HttpContext httpContext)
+        {
+            // We check Ordinal explicitly first because it's faster than OrdinalIgnoreCase
+            if (httpContext.Request.Path.StartsWithSegments(_path, StringComparison.Ordinal) ||
+                httpContext.Request.Path.StartsWithSegments(_path, StringComparison.OrdinalIgnoreCase))
+            {
+                httpContext.Response.StatusCode = 200;
+                httpContext.Response.ContentType = "application/json";
+                httpContext.Response.ContentLength = 30;
```
Comment (mgravell) [inline src/Benchmarks/JsonJilMiddleware.cs]: This is copied from JsonMiddleware.cs, I note - these hard-coded 30s smell a bit like cheating, though, no? (in both cases; I'm not limiting this to #36)

Comment (davidfowl) [inline src/Benchmarks/JsonJilMiddleware.cs]: Cheating why?

Comment (benaadams) [inline src/Benchmarks/JsonJilMiddleware.cs]: If you want to go fast you should be doing `httpContext.Response.Headers["Content-Length"] = "30";` as is done in the [plaintext middleware](https://github.com/aspnet/benchmarks/blob/master/src/Benchmarks/PlaintextMiddleware.cs#L34)

As the numeric form [does a bunch of conversions](https://github.com/aspnet/HttpAbstractions/issues/407)

Comment (mgravell) [inline src/Benchmarks/JsonJilMiddleware.cs]: @davidfowl it just seems odd to hard-code the content length, rather than either ask the serializer what it *will* be, or (more commonly) buffer the output, write the length of the buffer, then write the buffer. I realize the intent is to remove the buffer, but .... meh, maybe I'm over-thinking it.

Comment (NickCraver) [inline src/Benchmarks/JsonJilMiddleware.cs]: My only goal was to put the two on equal footing - I'm all for changing both to be more realistic, though.

File: src/Benchmarks/project.json (hunk 1 above)
Comment (NickCraver) [inline src/Benchmarks/project.json]: Yeah sure - I wasn't sure what the desire was since Jil's in beta for a bit while some missing corefx bits are added (waiting on https://github.com/dotnet/corefx/issues/4543 for another rev). I'll update to be specific though.

Comment (mgravell) [inline src/Benchmarks/project.json]: Everything except Newtonsoft.Json uses wildcard... just sayin'

There are pros and cons of either approach; it saved me having to issue a PR when I last rev'd "dapper" ;p

**Comment by DamianEdwards** [comment]
@NickCraver would you mind rebasing this so we can measure?

**Comment by NickCraver** [comment]
@DamianEdwards done!

**Comment by davidfowl** [comment]
This PR is stale now, needs a do-over.

**Comment by NickCraver** [comment]
I've paused RC2 work since things are still changing and we don't want to eat the time on the library side. I've ported Sigil (needed for Jil) but Jil still needs a fair amount of work for core. Given that many of the API differences that make it more difficult to port (and maintain performance) might be coming back...I'm personally inclined to leave it until that happens and I'm fairly sure Kevin's in the same boat.

That being said, I know Kevin would be happy to take an RC2 pull request on it. Due to RC1 tooling craziness current master doesn't support core at all (it was torn out to get a release out the door), so be aware it's pretty much a non-core to RC2 core port in the current state. I had a branch going for this, but to get Jil updated and run the performance tests I also need protobuf and we hit `coreclr.exe` crashing exceptions on that front. Theoretically, Jil is closer but so far away from an RC2 update on that branch it may be better to start over. Completing the set of libraries needed for testing is also non-trivial. I'm not sure if `SimplerSpeedTester` which Jil also uses is even planned for a port, that one also means changing all benchmark code which creates a conundrum of what comparisons are even valid. That'd have to be a cross-branch port for apples to apples ensuring no performance degradation.

TL;DR: I've done this PR twice now - don't have it in me to do it a third time. But, if someone else is willing to RC2-ify Jil then I'm happy to review it. If you want to close this out that's disappointing but may be best. It's a shame we never got a test run.
